Funcky.SourceGenerator/OrNoneFromTryPatternGenerator.cs

```diff
@@ -40,69 +37,134 @@ private static IncrementalValueProvider<ImmutableArray<MethodPartial>> GetOrNone
+            ? parsedType
+            : parsedType
+                .WithLeadingTrivia(Trivia(NullableDirectiveTrivia(Token(SyntaxKind.DisableKeyword), true)))
+                .WithTrailingTrivia(Trivia(NullableDirectiveTrivia(Token(SyntaxKind.EnableKeyword), true)));
     }
 
-    private static string GetMethodValue(Compilation compilation, MethodDeclarationSyntax method, AttributeSyntax attribute)
-        => attribute.ArgumentList?.Arguments.Count > 1 && attribute.ArgumentList?.Arguments[1].Expression is { } descriptionExpr
-            ? compilation
-                .GetSemanticModel(method.SyntaxTree)
-                .GetConstantValue(descriptionExpr)
-                .ToString()
-            : throw new Exception("Method value on attribute missing.");
-
-    private static TypeSyntax GetTypeValue(Compilation compilation, MethodDeclarationSyntax component, AttributeSyntax attribute)
-        => attribute.ArgumentList?.Arguments.Count > 0 && attribute.ArgumentList?.Arguments[0].Expression is TypeOfExpressionSyntax iconExpr
-            ? iconExpr.Type
-            : throw new Exception("Type value on attribute missing.");
-
-    private static string? GetClassName(SyntaxNode methodDeclaration)
-        => methodDeclaration
-            .TryGetParentSyntax<ClassDeclarationSyntax>()
-            ?.Identifier
-            .ToString();
-
-    private static string? GetNamespaceName(SyntaxNode methodDeclaration, Compilation compilation)
+    private static string GetNamespaceName(SyntaxNode methodDeclaration, Compilation compilation)
         => compilation.GetSemanticModel(methodDeclaration.SyntaxTree)
             .GetDeclaredSymbol(methodDeclaration)?
             .ContainingNamespace
-            .ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted));
+            .ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat.WithGlobalNamespaceStyle(SymbolDisplayGlobalNamespaceStyle.Omitted))
+                ?? throw new InvalidOperationException("Unable to detect containing namespace");
 
-    private static MethodDeclarationSyntax CreateMethodImplementation(MethodDeclarationSyntax methodDeclaration, string methodName, TypeSyntax typeSyntax)
-        => (MethodDeclarationSyntax)new OrNoneFromTryPatternRewriter(methodName, typeSyntax)
-            .Visit(methodDeclaration);
+    private static string GetMethodName(ITypeSymbol type, IMethodSymbol method)
+    {
+        const string tryPrefix = "Try";
+        const string orNoneSuffix = "OrNone";
+        return method.Name.StartsWith(tryPrefix)
+            ? $"{method.Name.Substring(tryPrefix.Length)}{type.Name}{orNoneSuffix}"
+            : $"{method.Name}{type.Name}{orNoneSuffix}";
+    }
 
-    private static string EmitCode(SyntaxNode syntaxNode)
-        => syntaxNode.ToFullString();
+    private static string GetParameterName(IParameterSymbol parameter, int index)
+        => index == 0
```

Comment:
not the is operator?

Reply:
[b05b317](https://github.com/polyadic/funcky/pull/634/commits/b05b31770786edd090edefacc6f9d71a91d165ed)